### PR TITLE
Handle socket(2)'s args (Fix #89)

### LINKED ISSUE #89: Handle socket(2)'s arguments
Some applications call `socket(2)` with `SOCK_CLOEXEC` or `SOCK_NONBLOCK`.
`getsockopt(fd, SOL_SOCKET, SO_TYPE, ...)` returns only its actual type `SOCK_STREAM` and these flags are ignored.
These flags must be checked and included in host-side socket creation.

```c
#include <stdio.h>
#include <string.h>
#include <unistd.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <arpa/inet.h>
#include <fcntl.h>

int main() {
    int sock_flags[3] = {SOCK_STREAM, SOCK_STREAM|SOCK_CLOEXEC, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK};
    char* sock_flags_str[3] = {"SOCK_STREAM", "SOCK_STREMA|SOCK_CLOEXEC", "SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK"};
    for (int i = 0; i < 3; i++) {
        int sock_flag = sock_flags[i];
        printf("socket %s flags=0x%x\n", sock_flags_str[i], sock_flags[i]);
        int sock = socket(AF_INET, sock_flag, 0);
        if (sock == -1) 
        {
            perror("socket");
            return -1;
        }

        int opt;
        int opt_len = sizeof(opt);
        int err = getsockopt(sock, SOL_SOCKET, SO_TYPE, &opt, &opt_len);
        if (err == -1)
        {
            perror("getsockopt");
            return -1;
        }
        printf("SOL_SOCKET SO_TYPE=0x%x\n", opt);

        int flags = fcntl(sock, F_GETFD);
        printf("F_GETFD flags=0x%x\n", flags);
        printf("CLOEXEC=%d\n", (flags & FD_CLOEXEC) > 0);
        flags = fcntl(sock, F_GETFL);
        printf("F_GETFL flags=0x%x\n", flags);
        printf("NON_BLOCK=%d\n", (flags & O_NONBLOCK) > 0);
        printf("\n");

        close(sock);
    }
}
```

```
$ ./check_sock
socket SOCK_STREAM flags=0x1
SOL_SOCKET SO_TYPE=0x1
F_GETFD flags=0x0
CLOEXEC=0
F_GETFL flags=0x2
NON_BLOCK=0

socket SOCK_STREMA|SOCK_CLOEXEC flags=0x80001
SOL_SOCKET SO_TYPE=0x1
F_GETFD flags=0x1
CLOEXEC=1
F_GETFL flags=0x2
NON_BLOCK=0

socket SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK flags=0x80801
SOL_SOCKET SO_TYPE=0x1
F_GETFD flags=0x1
CLOEXEC=1
F_GETFL flags=0x802
NON_BLOCK=1
```

### PR DESCRIPTION
**Pull request by naoki9911**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -208,10 +209,81 @@ jobs:
...
+    needs: test
     timeout-minutes: 60
     steps:
       - uses: actions/checkout@v4.1.3
-      - run: sudo ip a add 192.168.6.2/32 dev eth0
-      - run: hostname -I
-      - run: ./test/init_test.sh
-      - run: ~/bypass4netns/benchmark/run_bench.sh
+        # thanks to https://github.com/NobuoTsukamoto/my_actions_test/blob/main/.github/workflows/test_contains.yml
+      - name: cleanup
```
Comment (AkihiroSuda) [inline .github/workflows/test.yaml:217]: Why do you need to clean up the disk?
For ENOSPC?

File: test/init_test.sh [hunk 2]
```diff
@@ -12,8 +12,8 @@ if [ "$(whoami)" != "$TEST_USER" ]; then
     exit 0
 fi
 
-GO_VERSION="1.22.0"
-NERDCTL_VERSION="2.0.0-beta.3"
+GO_VERSION="1.23.2"
+NERDCTL_VERSION="2.0.0-rc.3"
```
Comment (AkihiroSuda) [inline test/init_test.sh:16]: Now GA

File: .github/workflows/test.yaml (hunk 1 above)
Comment (naoki9911) [reply .github/workflows/test.yaml:217]: Yes, due to ENOSPC. Without this, test images occupied GHA runner storage and  tests failed.

File: test/init_test.sh (hunk 2 above)
Comment (naoki9911) [reply test/init_test.sh:16]: Fixed.
Congratulations on nerdctl 2.0.0 GA! 🎉

Comment (AkihiroSuda) [reply test/init_test.sh:16]: The fix doesn't seem pushed?

Comment (naoki9911) [reply test/init_test.sh]: I fixed the wrong commit.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**